src/aggregator/aggregator/entry_test.go

```diff
@@ -1483,31 +1483,51 @@ func TestEntryAddTimedMetricTooLate(t *testing.T) {
 	e.opts = e.opts.SetBufferForPastTimedMetricFn(timedAggregationBufferPastFn)
 
 	inputs := []struct {
-		timeNanos     int64
-		storagePolicy policy.StoragePolicy
+		timeNanos       int64
```

Comment:
this test fails with the previous code.

this was a really subtle bug. with uber multierr, if there is only 1 error it's not a multierr, but the original type. in the case InnerError(errTooFarInPast). This would pass the m3errors.Is() check because it can explicitly handle an InnerError.

however, if there are multiple errors (e.g due to multiple storage errors), then the type is uber.multiError. In that case the m3.errors cannot handle probably handle an uber.multiError that contains m3.InnerErrors.

Ideally we would get rid of the m3 InnerErrors in favor of just implementing `func Is(err) bool` on every type of InnerError.